rrmngmnt/filesystem.py

```diff
@@ -226,3 +227,193 @@ def wget(self, url, output_file, progress_handler=None):
                 "Failed to download file from url {0}".format(url)
             )
         return output_file
+
+    def mktemp(self, template=None, tmpdir=None, directory=False):
+        """
+        Make temporary file
+
+        :param template: template for path, 'X's are replaced
+        :type template: str
+        :param tmpdir: where to create file, if not specified
+            use $TMPDIR if set, else /tmp
+        :type tmpdir: str
+        :param directory: create directory instead of a file
+        :type directory: bool
+        :return: absolute path to file or None if failed
+        :rtype: str
+        """
+        cmd = ['mktemp']
+        if tmpdir:
+            cmd.extend(['-p', tmpdir])
+        if directory:
+            cmd.append('-d')
+        if template:
+            cmd.append(template)
+        rc, out, _ = self.host.run_command(cmd)
+        if rc:
+            raise FailCreateTemp(cmd)
+        return out.replace('\n', '')
+
+    def mount_point(
+        self, source, target=None, fs_type=None, opts=None
+    ):
+        return MountPoint(
+            self,
+            source=source,
+            target=target,
+            fs_type=fs_type,
+            opts=opts,
+        )
+
+
+class MountPoint(Resource):
+    """
+    Class for mounting devices.
+    """
+    def __init__(self, fs, source, target=None, fs_type=None, opts=None):
+        """
+        Mounts source to target mount point
+
+        __author__ = "vkondula"
+        :param fs: FileSystem object instance
+        :type fs: FileSystem
+        :param source: Full path to source
+        :type source: str
+        :param target: Path to target directory, if omitted, a temporary
+        folder is created instead
+        :type target: str
+        :param fs_type: File system type
+        :type fs_type: str
+        :param opts: Mount options separated by a comma such as:
+        'sync,rw,guest'
+        :type opts: str
+        """
+        super(MountPoint, self).__init__()
+        self.fs = fs
+        self.source = source
+        self.opts = opts
+        self.fs_type = fs_type
+        self.target = target
+        self._tmp = not bool(target)
+        self._mounted = False
+
+    def __enter__(self):
+        try:
+            self.mount()
+        except MountError as e:
+            self.logger.error(e)
+            raise
+        return self
+
+    def __exit__(self, type_, value, tb):
+        try:
+            self.umount()
+        except MountError as e:
+            self.logger.error(e)
+            raise
```

Comment:
you will lose prospective exception which might happen in with_stm body ... try following code:

```python
class MyObject(object):
    def __enter__(self):
        print "__enter__"
        return self
    def __exit__(selfi, type_, value, tb):
        print "__exit__"
        raise Exception("ola")


if __name__ == "__main__":
    with MyObject() as mo:
        print "with_stm body"
        raise Exception("hello")
```